### PR TITLE
Fixes #2579 - be smarter in timing tests

### LINKED ISSUE #2579: Shaky Unit Test: PortWithRoleRandomPortsFromRangesTest is lazy with multiple ranges
This test seems to be shaky:

```
*** 1 TEST FAILED ***
PortWithRoleRandomPortsFromRangesTest:
- is lazy with multiple ranges *** FAILED *** (1 second, 79 milliseconds)
  org.scalatest.exceptions.TestFailedException was thrown. (PortWithRoleRandomPortsFromRangesTest.scala:99)
  org.scalatest.exceptions.TestFailedException:
  at org.scalatest.Assertions$class.newAssertionFailedException(Assertions.scala:498)
  at mesosphere.marathon.tasks.PortWithRoleRandomPortsFromRangesTest.newAssertionFailedException(PortWithRoleRandomPortsFromRangesTest.scala:15)
  at org.scalatest.Assertions$AssertionsHelper.macroAssert(Assertions.scala:394)
  at mesosphere.marathon.tasks.PortWithRoleRandomPortsFromRangesTest$$anonfun$6.apply$mcV$sp(PortWithRoleRandomPortsFromRangesTest.scala:99)
  at mesosphere.marathon.tasks.PortWithRoleRandomPortsFromRangesTest$$anonfun$6.apply(PortWithRoleRandomPortsFromRangesTest.scala:78)
  at mesosphere.marathon.tasks.PortWithRoleRandomPortsFromRangesTest$$anonfun$6.apply(PortWithRoleRandomPortsFromRangesTest.scala:78)
  at org.scalatest.Transformer$$anonfun$apply$1.apply$mcV$sp(Transformer.scala:22)
  at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
  at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
  at org.scalatest.Transformer.apply(Transformer.scala:22)
  at org.scalatest.Transformer.apply(Transformer.scala:20)
  at org.scalatest.FunSuiteLike$$anon$1.apply(FunSuiteLike.scala:158)
  at org.scalatest.Suite$class.withFixture(Suite.scala:1121)
  at mesosphere.marathon.tasks.PortWithRoleRandomPortsFromRangesTest.withFixture(PortWithRoleRandomPortsFromRangesTest.scala:15)
  at org.scalatest.FunSuiteLike$class.invokeWithFixture$1(FunSuiteLike.scala:155)
  at org.scalatest.FunSuiteLike$$anonfun$runTest$1.apply(FunSuiteLike.scala:167)
  at org.scalatest.FunSuiteLike$$anonfun$runTest$1.apply(FunSuiteLike.scala:167)
  at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
  at org.scalatest.FunSuiteLike$class.runTest(FunSuiteLike.scala:167)
  at mesosphere.marathon.tasks.PortWithRoleRandomPortsFromRangesTest.org$scalatest$BeforeAndAfter$$super$runTest(PortWithRoleRandomPortsFromRangesTest.scala:15)
  at org.scalatest.BeforeAndAfter$class.runTest(BeforeAndAfter.scala:200)
  at mesosphere.marathon.tasks.PortWithRoleRandomPortsFromRangesTest.runTest(PortWithRoleRandomPortsFromRangesTest.scala:15)
  at org.scalatest.FunSuiteLike$$anonfun$runTests$1.apply(FunSuiteLike.scala:200)
  at org.scalatest.FunSuiteLike$$anonfun$runTests$1.apply(FunSuiteLike.scala:200)
  at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:413)
  at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:401)
  at scala.collection.immutable.List.foreach(List.scala:381)
  at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
  at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:396)
  at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:483)
  at org.scalatest.FunSuiteLike$class.runTests(FunSuiteLike.scala:200)
  at mesosphere.marathon.tasks.PortWithRoleRandomPortsFromRangesTest.runTests(PortWithRoleRandomPortsFromRangesTest.scala:15)
  at org.scalatest.Suite$class.run(Suite.scala:1423)
  at mesosphere.marathon.tasks.PortWithRoleRandomPortsFromRangesTest.org$scalatest$FunSuiteLike$$super$run(PortWithRoleRandomPortsFromRangesTest.scala:15)
  at org.scalatest.FunSuiteLike$$anonfun$run$1.apply(FunSuiteLike.scala:204)
  at org.scalatest.FunSuiteLike$$anonfun$run$1.apply(FunSuiteLike.scala:204)
  at org.scalatest.SuperEngine.runImpl(Engine.scala:545)
  at org.scalatest.FunSuiteLike$class.run(FunSuiteLike.scala:204)
  at mesosphere.marathon.tasks.PortWithRoleRandomPortsFromRangesTest.org$scalatest$BeforeAndAfter$$super$run(PortWithRoleRandomPortsFromRangesTest.scala:15)
  at org.scalatest.BeforeAndAfter$class.run(BeforeAndAfter.scala:241)
  at mesosphere.marathon.tasks.PortWithRoleRandomPortsFromRangesTest.run(PortWithRoleRandomPortsFromRangesTest.scala:15)
  at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:444)
  at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:651)
  at sbt.ForkMain$Run$2.call(ForkMain.java:294)
  at sbt.ForkMain$Run$2.call(ForkMain.java:284)
  at java.util.concurrent.FutureTask.run(FutureTask.java:266)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
  at java.lang.Thread.run(Thread.java:745)
[info] Run completed in 53 seconds, 809 milliseconds.
[info] Total number of tests run: 740
[info] Suites: completed 124, aborted 0
[info] Tests: succeeded 739, failed 1, canceled 0, ignored 2, pending 0
[info] *** 1 TEST FAILED ***
[error] Failed tests:
[error] 	mesosphere.marathon.tasks.PortWithRoleRandomPortsFromRangesTest
[error] (marathon/test:test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 243 s, completed Nov 6, 2015 10:02:19 AM
The command "sbt clean coverage doc assembly" exited with 1.
```

### PR DESCRIPTION
**Pull request by kolloch**
for lazy port ranges.

Also call System.gc to prevent the GC from interfering.

### CODE REVIEW

**Comment by meichstedt** [comment]
`mesosphere.marathon.integration.InfoIntegrationTest` failed, but has nothing to do with this

File: src/test/scala/mesosphere/marathon/tasks/PortWithRoleRandomPortsFromRangesTest.scala [hunk 1]
```diff
@@ -63,25 +63,46 @@ class PortWithRoleRandomPortsFromRangesTest extends MarathonSpec {
...
+    // warm up
+    (1 to 100).foreach(_ => performTest())
+
+    // real test
+    System.gc()
+    val start = System.nanoTime()
+    performTest()
     val duration = FiniteDuration(System.nanoTime() - start, TimeUnit.NANOSECONDS)
+
+    val allowTimeForTest = benchmarkDuration
```
Comment (meichstedt) [inline src/test/scala/mesosphere/marathon/tasks/PortWithRoleRandomPortsFromRangesTest.scala]: I assume you wanted to use this value in the assert on line 97, correct? Then please go ahead and do so :)

Comment (kolloch) [inline src/test/scala/mesosphere/marathon/tasks/PortWithRoleRandomPortsFromRangesTest.scala]: ack!

**Comment by kolloch** [comment]
fixed that.

**Pull request merged**